.github/workflows/build_and_test.yml

```diff
@@ -8,28 +8,98 @@ on:
   pull_request:
 
 jobs:
-  build-and-test:
-    name: Build and Test Web and Electron Apps
-    runs-on: ubuntu-latest
+  linux:
+    name: Linux
+    runs-on: ubuntu-20.04
 
     steps:
-      - name: Checkout repo
+      - name: Checkout
         uses: actions/checkout@v2.3.4
 
-      - name: Install Node.js
+      - name: Install Node
         uses: actions/setup-node@v2.2.0
         with:
           node-version: 16
           cache: npm
       
-      - name: Install dependencies
+      - name: Install Dependencies
         run: npm ci
 
-      - name: Build web app
+      - name: Build Web App
         run: npx gulp build --platform=browser
-      
-      - name: Test web app
+
+      - name: Test Web App
         run: npm run action scripts/test
 
-      - name: Build Electron app
-        run: npm run action src/electron/build
+      - name: Build Electron App
+        run: npm run action src/electron/package_linux
+
+  windows:
+    name: Windows
+    runs-on: windows-2019
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Build Windows Client
+        run: npm run action src/electron/package_windows
+
+  mac:
+    name: Apple
+    runs-on: macos-10.15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Set XCode Settings
+        run: sudo xcode-select -switch /Applications/Xcode_11.7.app
```

Comment:
We may need a more recent version of Xcode.

@alalamav any idea why we are getting the Swift version mismatch?

Reply:
Apologies for my iOS ignorance, but it seems the Swift version is tied to the XCode version. Xcode 12.4 is the latest version that come pre-installed in github actions (who's ToS happens to be vetted by us) -

https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#xcode

who's latest is `5.3.2`; we need `5.4.2`

So i suppose we either need to recompile with an earlier compiler or manually install a new xcode version. I'll start with the latter

Reply:
I think it's probably less of a hassle to stick to what Github offers and recompile the dependency. It doesn't seem that complicated.

@alalama do we need xcode higher than 12.4?

Reply:
/cc @alalamav

Reply:
I updated the project to xcode 12.5 in #1046. That's the version I used to build the latest dependencies. I think it would be easier to update locally to xcode 12.5 than to rebuild dependencies and downgrade the project.